hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/BaseSparkCommitActionExecutor.java

```diff
@@ -292,8 +294,16 @@ protected void setCommitMetadata(HoodieWriteMetadata<HoodieData<WriteStatus>> re
   @Override
   protected void commit(HoodieWriteMetadata<HoodieData<WriteStatus>> result) {
     context.setJobStatus(this.getClass().getSimpleName(), "Commit write status collect: " + config.getTableName());
-    commit(result.getWriteStatuses(), result, result.getWriteStats().isPresent()
-        ? result.getWriteStats().get() : result.getWriteStatuses().map(WriteStatus::getStat).collectAsList());
+    if (!config.getBoolean(WRITE_IGNORE_FAILED) && result.getWriteStatuses().filter(status -> status.getErrors().size() > 0).count() > 0) {
+      WriteStatus writeStatus = result.getWriteStatuses().filter(status -> status.getErrors().size() > 0).collectAsList().get(0);
+      HashMap<HoodieKey, Throwable> errors = writeStatus.getErrors();
+      // get first one of error info to show error details
+      HoodieKey key = errors.keySet().iterator().next();
+      throw new HoodieException("Error writing record in the job", errors.get(key));
+    }
+    HoodieData<WriteStatus> filterResult = result.getWriteStatuses().filter(status -> status.getErrors().size() == 0);
+    commit(filterResult, result, result.getWriteStats().isPresent()
```

Comment:
Why filter out write status with errors?

Reply:
seems not need do this operation，I would revert it @beyond1920

Reply:
Yes.
Even though a file including some write errors, it might also including other valid records. Filters out those files is not reasonable.

Reply:
Yeah，+1